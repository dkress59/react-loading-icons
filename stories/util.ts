import { ArgTypes } from '@storybook/react'
import { SVGProps } from 'react'

export type IconProps = JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
export const IconFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: 'transparent',
	strokeOpacity: 1,
	width: 'auto',
}
export const IconStroke: IconProps = {
	fill: 'transparent',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}
export const IconStrokeFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}

export const IconFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: 'transparent',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: 'transparent',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
