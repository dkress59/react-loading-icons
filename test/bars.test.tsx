import { render, screen } from '@testing-library/react'
import React from 'react'

import Bars from '../src/components/bars'
import { testProps } from './util'

describe('<Bars /> component', () => {
	it('renders and matches snapshot', () => {
		render(<Bars data-testid="icon" />)
		const svg = screen.getByTestId('icon')
		expect(svg).not.toBeNull()
		expect(svg).toMatchSnapshot()
		expect(svg.getAttributeNames()).toContain('viewBox')
		expect(svg.getAttribute('class')).toBe('icon-loading')
	})
	it('inherits props', () => {
		render(<Bars {...{ ...testProps, speed: undefined }} />)
		const svg = screen.getByTestId('icon')
		expect(svg).toMatchSnapshot()
		expect(svg.getAttribute('height')).toBe('2em')
		expect(svg.getAttribute('width')).toBe('auto')
		expect(svg.getAttribute('class')).toBe('icon-loading test')
	})
})
