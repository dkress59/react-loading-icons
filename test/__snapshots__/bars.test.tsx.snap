// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bars /> component inherits props 1`] = `
<svg
  class="icon-loading test"
  data-testid="icon"
  height="2em"
  viewBox="0 0 135 140"
  width="auto"
>
  <rect
    fill="#000"
    fill-opacity=".5"
    height="120"
    rx="6"
    stroke="#fff"
    stroke-opacity=".5"
    width="15"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#000"
    fill-opacity=".5"
    height="120"
    rx="6"
    stroke="#fff"
    stroke-opacity=".5"
    width="15"
    x="30"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#000"
    fill-opacity=".5"
    height="140"
    rx="6"
    stroke="#fff"
    stroke-opacity=".5"
    width="15"
    x="60"
  >
    <animate
      attributeName="height"
      begin="0s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#000"
    fill-opacity=".5"
    height="120"
    rx="6"
    stroke="#fff"
    stroke-opacity=".5"
    width="15"
    x="90"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#000"
    fill-opacity=".5"
    height="120"
    rx="6"
    stroke="#fff"
    stroke-opacity=".5"
    width="15"
    x="120"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
</svg>
`;

exports[`<Bars /> component renders and matches snapshot 1`] = `
<svg
  class="icon-loading"
  data-testid="icon"
  height="140"
  viewBox="0 0 135 140"
  width="135"
>
  <rect
    fill="#fff"
    height="120"
    rx="6"
    width="15"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#fff"
    height="120"
    rx="6"
    width="15"
    x="30"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#fff"
    height="140"
    rx="6"
    width="15"
    x="60"
  >
    <animate
      attributeName="height"
      begin="0s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#fff"
    height="120"
    rx="6"
    width="15"
    x="90"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.25s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
  <rect
    fill="#fff"
    height="120"
    rx="6"
    width="15"
    x="120"
    y="10"
  >
    <animate
      attributeName="height"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="120;110;100;90;80;70;60;50;40;140;120"
    />
    <animate
      attributeName="y"
      begin="0.5s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="10;15;20;25;30;35;40;45;50;0;10"
    />
  </rect>
</svg>
`;
